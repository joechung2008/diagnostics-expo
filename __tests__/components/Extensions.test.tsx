import { render } from '@testing-library/react-native';
import Extensions from '../../components/Extensions';

// Mock the ThemeContext
jest.mock('../../components/ThemeContext', () => ({
  useTheme: () => ({
    colors: {
      background: '#ffffff',
      surface: '#f5f5f5',
      text: '#000000',
      border: '#cccccc',
    },
  }),
}));

// Mock the utils functions
jest.mock('../../lib/utils', () => ({
  byKey: jest.fn((a, b) => a.key.localeCompare(b.key)),
  isExtensionInfo: jest.fn(
    (value) => value && typeof value === 'object' && 'extensionName' in value
  ),
  toNavLink: jest.fn(({ extensionName }) => ({
    key: extensionName,
    name: extensionName,
    url: '',
  })),
}));

describe('Extensions', () => {
  const mockExtensions = {
    'extension-1': {
      extensionName: 'Extension A',
      manageSdpEnabled: true,
      config: { key1: 'value1' },
    },
    'extension-2': {
      extensionName: 'Extension B',
      manageSdpEnabled: false,
      stageDefinition: { dev: ['server1'] },
    },
    'extension-3': {
      extensionName: 'Extension C',
      manageSdpEnabled: true,
    },
  };

  const mockOnLinkClick = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders extensions list correctly', () => {
    const { toJSON } = render(
      <Extensions extensions={mockExtensions} onLinkClick={mockOnLinkClick} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders empty extensions list', () => {
    const { toJSON } = render(
      <Extensions extensions={{}} onLinkClick={mockOnLinkClick} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('filters out non-extension info items', () => {
    const extensionsWithErrors = {
      ...mockExtensions,
      'error-extension': {
        lastError: {
          errorMessage: 'Something went wrong',
          time: '2023-01-01T00:00:00Z',
        },
      },
    };

    const { toJSON } = render(
      <Extensions
        extensions={extensionsWithErrors}
        onLinkClick={mockOnLinkClick}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('handles single extension', () => {
    const singleExtension = {
      'single-ext': {
        extensionName: 'Single Extension',
        manageSdpEnabled: true,
      },
    };

    const { toJSON } = render(
      <Extensions extensions={singleExtension} onLinkClick={mockOnLinkClick} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with accessibility label', () => {
    const { toJSON } = render(
      <Extensions extensions={mockExtensions} onLinkClick={mockOnLinkClick} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
