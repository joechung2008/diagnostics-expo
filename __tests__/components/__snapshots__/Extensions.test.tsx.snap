// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extensions filters out non-extension info items 1`] = `
<RCTScrollView
  accessibilityLabel="Extensions"
  style={
    {
      "backgroundColor": "#ffffff",
      "flexGrow": 0,
      "flexShrink": 1,
      "minWidth": 0,
      "padding": 10,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Extension A
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Extension B
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Extension C
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Extensions handles single extension 1`] = `
<RCTScrollView
  accessibilityLabel="Extensions"
  style={
    {
      "backgroundColor": "#ffffff",
      "flexGrow": 0,
      "flexShrink": 1,
      "minWidth": 0,
      "padding": 10,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Single Extension
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Extensions renders empty extensions list 1`] = `
<RCTScrollView
  accessibilityLabel="Extensions"
  style={
    {
      "backgroundColor": "#ffffff",
      "flexGrow": 0,
      "flexShrink": 1,
      "minWidth": 0,
      "padding": 10,
    }
  }
>
  <View />
</RCTScrollView>
`;

exports[`Extensions renders extensions list correctly 1`] = `
<RCTScrollView
  accessibilityLabel="Extensions"
  style={
    {
      "backgroundColor": "#ffffff",
      "flexGrow": 0,
      "flexShrink": 1,
      "minWidth": 0,
      "padding": 10,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Extension A
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Extension B
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Extension C
      </Text>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Extensions renders with accessibility label 1`] = `
<RCTScrollView
  accessibilityLabel="Extensions"
  style={
    {
      "backgroundColor": "#ffffff",
      "flexGrow": 0,
      "flexShrink": 1,
      "minWidth": 0,
      "padding": 10,
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Extension A
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Extension B
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f5f5f5",
          "borderColor": "#cccccc",
          "borderRadius": 4,
          "borderWidth": 1,
          "justifyContent": "center",
          "marginBottom": 8,
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Extension C
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
